refactor(semantic): rename `cfg` macro to `control_flow`.

rzvxa opened the pull request:
As requested here https://github.com/oxc-project/oxc/pull/3737#discussion_r1644713121

rzvxa commented:
* **#3742** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3742?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented:
Too much variance coming from a small benchmark file, I'll look into it tomorrow.

rzvxa committed:
refactor(semantic): rename `cfg` macro to `control_flow`. (#3742)

As requested here https://github.com/oxc-project/oxc/pull/3737#discussion_r1644713121

overlookmotel commented:
> Too much variance coming from a small benchmark file, I'll look into it tomorrow.

My guess is the variance is coming from `SourcemapBuilder::generate_line_offset_tables` which builds a large `Vec`. There's a large `Vec` reallocation (`RawVec::grow_one` followed by memcpy) at start of the flamegraph. If by chance the global allocator says that `Vec` can grow in place, then the memcpy would disappear, which would be a large jump in perf. But whether that happens or not will be non-deterministic depending on state of global allocator (or even the OS).

That's just a guess, though...
